jfr-daemon/src/main/java/com/newrelic/jfr/daemon/agent/AgentMain.java

```diff
@@ -35,6 +37,12 @@ public static void premain(String agentArgs, Instrumentation inst) {
   private void start() {
     DaemonConfig config = SetupUtils.buildConfig();
     Attributes commonAttrs = SetupUtils.buildCommonAttributes();
+    // When running as a standalone agent, jfr needs to set service name because it can't obtain an
+    // entity guid.
+    // Without the service name, ingest will not tag data with an entity guid and it will be
+    // difficult to query.
+    commonAttrs.put(SERVICE_NAME, config.getMonitoredAppName());
```

Comment:
Nice, so this fixes the "orphaned" jfr event/metric issue that we were seeing? We'll probably want to also set service name as a common attribute in the JFRService, for consistency.

Reply:
Here is quick change for the jfr service in the agent to address this

https://github.com/newrelic/newrelic-java-agent/pull/263

Reply:
Nice, yea. I see that in `JFRDaemon.main` we add both the `app.name` and `service.name` attributes the same way that you did in the PR for the agent. We should make sure to also add `app.name` in `AgentMain`, so that every use case has both attributes. Since all three use cases (jfr service, jfr agent, jfr daemon) will have both of these attributes set from the `DaemonConfig`, I wonder if we can't just pull them into the `buildCommonAttributes` method as a common place?

One thing I don't quite understand is why, in `JFRDaemon.main`, so we only set them if the `entity.guid` can't be obtained? Is there a problem if we set the `entity.guid` and name attributes for the daemon?

```
                if (optGuid.isPresent()) {
                  commonAttrs.put(ENTITY_GUID, optGuid.get());
                } else {
                  commonAttrs.put(APP_NAME, config.getMonitoredAppName());
                  commonAttrs.put(SERVICE_NAME, config.getMonitoredAppName());
                }

```

Reply:
I pushed a WIP commit that always adds service and app name to the common attributes. It does it in one place now, SetupUtils.

Here is the reported data for the standalone. I intentionally set the name with NEW_RELIC_APP_NAME

<img width="674" alt="Screen Shot 2021-04-05 at 4 08 19 PM" src="https://user-images.githubusercontent.com/24482401/113637727-29dbde00-962a-11eb-9da7-3cfc6c93bc39.png">

Here is reported data, with the daemon embedded into the agent. No env for NEW_RELIC_APP_NAME was set.

<img width="605" alt="Screen Shot 2021-04-05 at 4 05 42 PM" src="https://user-images.githubusercontent.com/24482401/113637757-3b24ea80-962a-11eb-949f-434fef812b2c.png">


Here is the reported data, with the daemon running as a separate process to the java agent process. No env for NEW_RELIC_APP_NAME was set.

<img width="605" alt="Screen Shot 2021-04-05 at 4 04 27 PM" src="https://user-images.githubusercontent.com/24482401/113637811-5bed4000-962a-11eb-9867-b3c02403513e.png">

Reply:
>One thing I don't quite understand is why, in JFRDaemon.main, we only set them if the entity.guid can't be obtained? Is there a problem if we set both the entity.guid and the app/service name attributes for the daemon?

It might be a little confusing. 

Unless the customer manually specifies app name when they run the daemon jar, they will have JFR data with the same entity guid, but with name of "My Application"